index.js

```diff
@@ -73,32 +73,30 @@ class USHINBase {
       );
     }
 
-    const allPoints = new Set();
-
-    if (main) allPoints.add(main);
-
-    for (const shape of Object.keys(shapes)) {
-      const pointIds = shapes[shape];
-      for (const pointId of pointIds) {
-        const point = pointStore[pointId];
-        if (!point) {
-          const error = new Error("Point ID not found in store");
-          error.pointId = pointId;
-          throw error;
-        }
-        if (!point._id) throw new Error("Must specify point ID");
-        if (!point._rev) {
-          await this.addPoint({ createdAt: createdAtTime, ...point });
-        }
-        allPoints.add(pointId);
-        if (point.referenceHistory) {
-          for (const { pointId: referencePoint } of point.referenceHistory) {
-            allPoints.add(referencePoint);
-          }
-        }
+    const allPoints = new Set([main, ...Object.values(shapes).flat()]);
```

Comment:
Do we have something guaranteeing that `main` isn't `null` or `undefined`?

Reply:
Good catch! I thought I was getting all fancy with the concise syntax XD

Reply:
Actually, if main doesn't exist, we should throw an error. Shall I add that at the very beginning of the function (before `const { authorURL } = this;`)?